Tests/IntegrationTests/PSQLIntegrationTests.swift

```diff
@@ -212,6 +212,31 @@ final class IntegrationTests: XCTestCase {
         XCTAssertEqual(try result?.rows.first?.decode([Double].self, context: .default), doubles)
     }
 
+    func testDoubleArraySerializationQueryWithCollectionInterpolation() {
+        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
+        defer { XCTAssertNoThrow(try eventLoopGroup.syncShutdownGracefully()) }
+        let eventLoop = eventLoopGroup.next()
+
+        var conn: PostgresConnection?
+        XCTAssertNoThrow(conn = try PostgresConnection.test(on: eventLoop).wait())
+        defer { XCTAssertNoThrow(try conn?.close().wait()) }
+
+        var result: PostgresQueryResult?
+        let doubles: [Double] = [3.14, 42]
+        XCTAssertNoThrow(try conn?.query("CREATE TABLE foo (accuracy double precision)", logger: .psqlTest).wait())
+        XCTAssertNoThrow(try conn?.query("INSERT INTO foo VALUES (\(doubles[0]))", logger: .psqlTest).wait())
+        XCTAssertNoThrow(try conn?.query("INSERT INTO foo VALUES (\(doubles[1]))", logger: .psqlTest).wait())
+        XCTAssertNoThrow(result = try conn?.query("SELECT * FROM foo WHERE accuracy IN \(collection: doubles)", logger: .psqlTest).wait())
```

Comment:
Writing the test like the above function fails because postgres returns the values as the RECORD type.
Anyway, this test much better demonstrates what i actually wanted this new function to do.